optuna/storages/rdb/models.py

```diff
@@ -176,6 +176,13 @@ def find_by_id(cls, trial_id, session):
 
         return trial
 
+    @classmethod
+    def delete_running_trials(cls, session):
+        # type: (orm.Session) -> int
+
+        n_trials_deleted = session.query(cls).filter(cls.state == TrialState.RUNNING).delete()
```

Comment:
I prefer state change approach to trial deletion because allowing deletion could introduce extra complexity to the Optuna code (e.g., we need to modify RDBStorage's caching mechanism if trial deletion should be supported).
Especially, [the current trial numbering method of `RDBStorage`](https://github.com/optuna/optuna/blob/52efa91beec06efd4f1453dfe4488f3fbf8ba9b7/optuna/storages/rdb/models.py#L250) strongly relies on the assumption that trials never be deleted. If this assumption can no longer be retained, we need to consider another way for numbering trials. I think that this could be a very difficult task.

Reply:
Agreed. I'll update the method.